describe/AbstractDataAttribute.ts

```diff
@@ -0,0 +1,112 @@
+import { EnumType, EnumTypeDescription } from "./EnumType.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { ValDescription, describeVal } from "./Val.js";
+
+export function isAbstractDataAttributeDescription(
+  type: AbstractDataAttributeDescription
+): type is AbstractDataAttributeDescription {
+  return (
+    (type as AbstractDataAttributeDescription).bType !== undefined &&
+    (type as AbstractDataAttributeDescription).count !== undefined &&
+    (type as AbstractDataAttributeDescription).valImport !== undefined &&
+    (type as AbstractDataAttributeDescription).val !== undefined &&
+    (type as AbstractDataAttributeDescription).valKind !== undefined
+  );
+}
+
+export interface AbstractDataAttributeDescription extends NamingDescription {
+  /** Required (B)DA bType attribute sAddr */
+  bType: string;
+  /** Reference to DAType (bType: Struct) or EnumType(bType: Enum) */
+  type?: DATypeDescription | EnumTypeDescription;
+  /** (B)DA sAddr attribute sAddr */
+  sAddr?: string;
+  /** (B)DA valKind attribute defaulting to "Set" */
+  valKind: "Spec" | "Conf" | "RO" | "Set";
+  /** (B)DA valKind attribute defaulting false */
+  valImport: boolean;
+  /** (B)DA valKind attribute defaulting 0 */
+  count: number;
+  val: ValDescription[];
+}
+
+/** Get count from referenced sibling element */
+function siblingCount(element: Element, reference: string): number {
+  const parent = element.parentElement;
+  if (!parent) return NaN;
+
+  const sibling = Array.from(parent.children).find(
+    (child) => child.getAttribute("name") === reference
+  );
+  if (!sibling) return NaN;
+
+  const count = sibling.getAttribute("count");
+  if (!count) return NaN;
+
+  return parseInt(count, 10);
+}
+
+export function describeAbstractDataAttribute(
+  element: Element
+): AbstractDataAttributeDescription {
+  const abstractDataAttributeDesc: AbstractDataAttributeDescription = {
+    ...describeNaming(element),
+    bType: element.getAttribute("bType") || "undefined",
+    valKind: "Set",
+    valImport: false,
+    count: 0,
+    val: [],
+  };
+
+  const [sAddr, valKind, valImport, type, count] = [
+    "sAddr",
+    "valKind",
+    "valImport",
+    "type",
+    "count",
+  ].map((attr) => element.getAttribute(attr));
+
+  if (sAddr) abstractDataAttributeDesc.sAddr = sAddr;
+
+  if (valKind && ["Spec", "Conf", "RO", "Set"].includes(valKind))
+    abstractDataAttributeDesc.valKind = valKind as
+      | "Spec"
+      | "RO"
+      | "Conf"
+      | "Set";
+
+  if (valImport && valImport === "true")
+    abstractDataAttributeDesc.valImport = true;
+
+  if (count && !isNaN(parseInt(count, 10)))
+    // count can be an unsigned integer
+    abstractDataAttributeDesc.count = parseInt(count, 10);
+  else if (count && !isNaN(siblingCount(element, count)))
+    // count can be a reference to another sibling that has integer definition
+    abstractDataAttributeDesc.count = siblingCount(element, count);
```

Comment:
Maybe only parse these once and use them later?

```suggestion
  // count can be an unsigned integer
  let countInt = parseInt(count, 10);
  // or count can be a reference to another sibling that has integer definition
  if (isNaN(countInt)) countInt = siblingCount(element, count);
  if (!isNaN(countInt)) abstractDataAttributeDesc.count = countInt;
```